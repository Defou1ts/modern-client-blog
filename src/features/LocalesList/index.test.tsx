import React from 'react';

import { render, screen } from '@testing-library/react';
import { useRouter } from 'next/router';

import renderer from 'react-test-renderer';

import { LocalesList } from './index';

import type { NextRouter } from 'next/router';

jest.mock('next/router', () => ({
	useRouter: jest.fn(),
}));

describe('LocalesList', () => {
	beforeAll(() => {
		const useRouterMock = {
			locale: 'en',
			locales: ['en', 'ru'],
		};
		(useRouter as jest.Mock<NextRouter>).mockReturnValue(useRouterMock as NextRouter);
	});

	it('renders a LocalesList', () => {
		render(<LocalesList />);

		const titleText = screen.getByText('en');
		expect(titleText).toBeInTheDocument();
		expect(titleText).toBeVisible();
	});

	it('button header snapshot', () => {
		const component = renderer.create(<LocalesList />);

		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
