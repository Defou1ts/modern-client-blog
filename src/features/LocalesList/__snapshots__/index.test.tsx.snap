// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocalesList button header snapshot 1`] = `
<div
  className="locales"
  data-test-id="locales"
>
  <a
    className="locale active"
    data-test-id="en"
    href="/"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onTouchStart={[Function]}
  >
    en
  </a>
  <a
    className="locale"
    data-test-id="ru"
    href="/"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onTouchStart={[Function]}
  >
    ru
  </a>
</div>
`;
